tests/foreman/cli/test_host.py

```diff
@@ -1745,6 +1751,186 @@ def test_negative_add_parameter(self):
                 self.host = Host.info({'id': self.host['id']})
                 self.assertNotIn(name, self.host['parameters'].keys())
 
+    @tier3
+    def test_negative_view_parameter_by_non_admin_user(self):
+        """Attempt to view parameters with non admin user without Parameter
+         permissions
+
+        :id: 65ba89f0-9bee-43d9-814b-9f5a194558f8
+
+        :steps:
+            1. As admin user create a host
+            2. Set a host parameter name and value
+            3. Create a non admin user with the following permissions:
+                Host: [view_hosts],
+                Organization: [view_organizations],
+            4. Get the host info as the non admin user
+
+        :expectedresults: The non admin user is not able to read the parameters
+
+        :BZ: 1296662
+
+        :CaseImportance: System
+        """
+        param_name = gen_string('alpha').lower()
+        param_value = gen_string('alphanumeric')
+        user_name = gen_string('alphanumeric')
+        user_password = gen_string('alphanumeric')
+        Host.set_parameter({
+            'host-id': self.host['id'],
+            'name': param_name,
+            'value': param_value,
+        })
+        host = Host.info({'id': self.host['id']})
+        self.assertEqual(host['parameters'][param_name], param_value)
+        role = make_role()
+        add_role_permissions(
+            role['id'],
+            resource_permissions={
+                'Host': {'permissions': ['view_hosts']},
+                'Organization': {'permissions': ['view_organizations']},
+            }
+        )
+        user = make_user({
+            'admin': False,
+            'default-organization-id': self.org_id,
+            'organization-ids': [self.org_id],
+            'default-location-id': self.loc_id,
+            'location-ids': [self.loc_id],
+            'login': user_name,
+            'password': user_password,
+        })
+        User.add_role({'id': user['id'], 'role-id': role['id']})
+        host = Host.with_user(
+            username=user_name,
+            password=user_password
+        ).info({'id': self.host['id']})
+        self.assertEqual(host['parameters'], {})
```

Comment:
so no error is returned?
Hmm, then such equal statement is not a prove for that functionality as you can receive empty dict even for admin user.
I understand that your next case cover that idea, but anyway

Reply:
if there are no better way to check - then just ignore my comment

Reply:
@oshtaier all the parameters area listed with entity info and there no other cmd to list them,

Reply:
Why don't we get an error? Like permission denied?

Reply:
This is not the purpose of the functionality, The user should work with host as usual (without errors) and even do not guess that there is some parameters that was hidden,
as the user is not reading the parameters explicitly but working with the host.

Reply:
The permission error should raise,  if the user try to edit/create/delete a param when it has no permission to do so.

Reply:
ACK to what @ldjebran said. However, BZ verification message says in case user don't have `view_params` permission - `parameters` section is not returned at all. The test results you've posted, are they final? Behavior could have changed in a year, that happens, i'm not surprised, but still would like to double check.

Reply:
Had a small chat and agreed `self.assertFalse(host.get('parameters'))` is a better choice here which covers both cases.